Mathlib/NumberTheory/Ostrowski.lean

```diff
@@ -0,0 +1,58 @@
+/-
+Copyright (c) 2024 María Inés de Frutos-Fernández. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: David Kurniadi Angdinata, Fabrizio Barroero, Laura Capuano, Nirvana Coppola,
+María Inés de Frutos Fernández, Sam van Gool, Silvain Rideau-Kikuchi, Amos Turchet,
+Francesco Veneziano
+-/
+
+import Mathlib.Analysis.Normed.Field.Basic
+import Mathlib.Analysis.SpecialFunctions.Pow.Real
+import Mathlib.Analysis.Normed.Ring.Seminorm
+
+/-!
+# Ostrowski’s Theorem
+
+The goal of this file is to prove Ostrowski’s Theorem which gives a list of all the nontrivial
+absolute values on a number field up to equivalence. (TODO)
+
+## References
+* [K. Conrad, *Ostroski's Theorem for Q*][conradQ]
+* [K. Conrad, *Ostroski for number fields*][conradnumbfield]
+* [J. W. S. Cassels, *Local fields*][cassels1986local]
+
+
+## Tags
+ring_norm, ostrowski
+-/
+
+namespace Rat.MulRingNorm
+open Int
+
+variable {f g : MulRingNorm ℚ}
+
+/-- Values of a multiplicative norm of the rationals coincide on ℕ if and only if they coincide
+on ℤ. -/
+lemma eq_on_Nat_iff_eq_on_Int : (∀ n : ℕ , f n = g n) ↔ (∀ n : ℤ , f n = g n) := by
+  refine ⟨fun h z ↦ ?_, fun a n ↦ a n⟩
+  obtain ⟨n , rfl | rfl⟩ := eq_nat_or_neg z <;>
+  simp only [Int.cast_neg, Int.cast_natCast, map_neg_eq_map, h n]
+
+/-- Values of a multiplicative norm of the rationals are determined by the values on the natural
+numbers. -/
+lemma eq_on_Nat_iff_eq : (∀ n : ℕ , f n = g n) ↔ f = g := by
+  refine ⟨fun h ↦ ?_, fun h n ↦ congrFun (congrArg DFunLike.coe h) ↑n⟩
+  ext z
+  rw [← Rat.num_div_den z, map_div₀, map_div₀, h, eq_on_Nat_iff_eq_on_Int.mp h]
+
+/-- The equivalence class of a multiplicative norm on the rationals is determined by its values on
+the natural numbers. -/
+lemma equiv_iff_equiv_on_Nat : (∃ c : ℝ, 0 < c ∧ (∀ n : ℕ , (f n)^c = g n)) ↔
```

Comment:
I think that also this name should be inverted, right? The `equiv` is the RHS in the statement but appears on the left in the name.

Reply:
Right. Thanks! Do you know if other than implementing all the bot's changes I have anything else to do? Or will it stop complaining eventually?